src/Thunks/api-ms-win-core-processthreads.hpp

```diff
@@ -158,6 +158,44 @@ namespace YY
 #endif
 
 
+#if (YY_Thunks_Support_Version < NTDDI_WS03)
+
+		//Windows Vista, Windows XP Professional x64 Edition [desktop apps only]
+		//Windows Server 2008, Windows Server 2003 with SP1 [desktop apps only]
+		__DEFINE_THUNK(
+		kernel32,
+		4,
+		BOOL,
+		WINAPI,
+		SetThreadStackGuarantee,
+			_Inout_ ULONG *StackSizeInBytes
+		)
+		{
+			if (auto pSetThreadStackGuarantee = try_get_SetThreadStackGuarantee())
+			{
+				return pSetThreadStackGuarantee(StackSizeInBytes);
+			}
+			else
+			{
+				ULONG prev_size = ((TEB*)NtCurrentTeb())->GuaranteedStackBytes;
+				ULONG new_size = (*StackSizeInBytes + 4095) & ~4095;
+
+				/* at least 2 pages on 64-bit */
+				if (sizeof(void*) > sizeof(int) && new_size) new_size = max(new_size, 8192);
```

Comment:
8192 = 4096 * 2 = GetSystemInfo().dwPageSize * 2

Reply:
@mingkuang-Chuyu The implementation from kernel32.dll in Windows Server 2003 Enterprise x86 with Service Pack 1 and the implementation from WoW64's kernel32.dll in Windows Server 2003 Enterprise x64 with Service Pack 2 **both** have the same logic. So, I don't think "at least 2 pages only on 64-bit" .